### PR TITLE
F add file type filter to s3 downloader

### PR DESCRIPTION
**Pull request by carebare47**
## Proposed changes

add file type exclusion list to s3 downloader

## Checklist

Before posting a PR ensure that from each of the below categories **AT LEAST ONE BOX HAS BEEN CHECKED**. If more than one category is applicable then more can be checked. Also ensure that the proposed changes have been filled out with relevant information for reviewers.

## Tests

- [ ] No tests required to be added. (For small changes that will be tested by CI/CD infrastructure).
- [ ] Added/Modified automated and PhantomHand CI tests (if a new class is added (Python or C++), the interface of that class must be unit tested).
- [ ] Manually tested in simulation (if simulation specific or no hardware required to test the functionality). 
- [ ] Manually tested on hardware (if hardware specific or related).

## Documentation

- [ ] No documentation required to be added.
- [ ] Added documentation (For any new feature, explain what it does and how to use it. Write the documentation in a relevant space, e.g. Github, Confluence, etc).
- [ ] Updated documentation (For changes to pre-existing features mentioned in the documentation).

### CODE REVIEW

File: sr_utilities_common/src/sr_utilities_common/aws_manager.py [hunk 1]
```diff
@@ -110,22 +111,59 @@ def _prepare_structure_download(self, files_base_path, files_folder_path, file_n
             else:
                 self.aws_paths.append(f"{file_name}")
 
-    def download(self, bucket_name, files_base_path, files_folder_path, file_names, bucket_subfolder):
+    def download(self, bucket_name, files_base_path, files_folder_path,
+                 file_names, bucket_subfolder, preserve_downloaded_folder_structure=False,
+                 file_extentions_to_exclude=None):
+        if file_extentions_to_exclude is not None:
+            examples = "(e.g. ['.txt', '.jpg'])"
```
Comment (niko-holmes) [inline sr_utilities_common/src/sr_utilities_common/aws_manager.py:118]: May be better to call the var `EXAMPLES` as it is a constant

File: sr_utilities_common/src/sr_utilities_common/aws_manager.py [hunk 2]
```diff
@@ -110,22 +111,59 @@ def _prepare_structure_download(self, files_base_path, files_folder_path, file_n
...
+                                             total=len(self.file_full_paths)):
+            if preserve_downloaded_folder_structure:
+                if not os.path.exists(os.path.join(directory, aws_path)):
+                    os.makedirs(os.path.join(directory, aws_path))
+                file_full_path = os.path.join(files_base_path, files_folder_path, aws_path)
+                if os.path.isdir(file_full_path):
+                    os.rmdir(file_full_path)
+            if file_extentions_to_exclude is not None:
+                if any(file_full_path.endswith(ext) for ext in file_extentions_to_exclude):
+                    ext = file_full_path.split('.')[-1]
```
Comment (niko-holmes) [inline sr_utilities_common/src/sr_utilities_common/aws_manager.py]: nice, didn't know you could do this.

**Review by niko-holmes — CHANGES_REQUESTED** [review]

File: sr_utilities_common/src/sr_utilities_common/aws_manager.py (hunk 1 above)
Comment (carebare47) [reply sr_utilities_common/src/sr_utilities_common/aws_manager.py:118]: not sure our linter would like that

Comment (niko-holmes) [reply sr_utilities_common/src/sr_utilities_common/aws_manager.py:118]: fair point

**Review by niko-holmes — APPROVED** [review]

**Review by dg-shadow — APPROVED** [review]

**Pull request merged**